### PR TITLE
Built CRUD operations on Grishma's work.

### PR DESCRIPTION
**Pull request by johnsciutto**
Added instance methods and static methods to use CRUD operations with the user database.

Did my best to build on Grishma's work.

Unit testing for CRUD operations is not present yet (working on it). I just wanted to make sure this (merging into someone else's branch) was working before adding more work.

### CODE REVIEW

File: server/userModel/user.js [hunk 1]
```diff
@@ -1,13 +1,196 @@
...
+/**
+ * Soft delete the current user instance.
+ *
+ * @returns {boolean|mongoose.Query} false if operation failed.
+ * @method
+ */
+userSchema.methods.softDelete = function () {
+  return User.updateOne(
+    { email: this.email },
+    { isActive: false },
```
Comment (jessdesr) [inline server/userModel/user.js]: The user schema does not have isActive defined: you can get around needing to add this to every schema in the database by making a soft deletion plugin!

**Review by jessdesr — COMMENTED** [review]
Great, we should hold off on merging this into Grishma's branch if her PR is going to be up soon @shahgrishma123 ?

File: server/userModel/user.js (hunk 1 above)
Comment (johnsciutto) [reply server/userModel/user.js]: Right! I forgot to add that. Thanks for catching it.

Cool, I didn't know that existed. I'll research how to do that.
Should I also add the property manually to the schema in the file?

Comment (johnsciutto) [reply server/userModel/user.js]: Resolved. Pushing now.

**Review by jessdesr — APPROVED** [review]

**Comment by jessdesr** [comment]
You'll need to resolve these conflicts, this is most likely due to the linting that was performed on your branch

**Pull request merged**